ruia/spider.py

```diff
@@ -501,11 +497,8 @@ async def start_master(self):
             self.logger.info(f"Worker started: {id(worker)}")
         await self.request_queue.join()
 
-        if not self.is_async_start:
+        if self.cancel_tasks:
             await self.stop(SIGINT)
-        else:
-            if self.cancel_tasks:
```

Comment:
`Spider.stop()` is basically `Spider.cancel_all_tasks()` with an additional `logger.info()` statement. Remove the `else` branch cause `is_async_start` is not used.